pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -695,6 +754,141 @@ exit 0`,
+		if !isCustomFileDestinationGood(finalDestination) {
+			return nil, fmt.Errorf("destination '%s' is bad", finalDestination)
+		}
+
+		files = append(files, map[string]string{
+			"source":      strings.TrimPrefix(customFile.Source, "/"),
+			"destination": finalDestination,
+		})
+	}
+
+	const copyCommandTemplate = `echo "Copying custom files..."
+{{- range $file := . }}
+mkdir -p "/custom/$(dirname '{{ $file.destination }}')"
+if [ -d "/{{ $file.source }}" ]; then
+  cp -TRv "/{{ $file.source }}" "/custom/{{ $file.destination }}"
+else
+  cp -Rv "/{{ $file.source }}" "/custom/{{ $file.destination }}"
+fi
+{{ end }}
+exit 0
+`
+	var copyCommand strings.Builder
+	t, err := template.New("tmpl").Funcs(map[string]any{"contains": strings.Contains}).Parse(copyCommandTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse template: %w", err)
+	}
+	err = t.Execute(&copyCommand, files)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute template: %w", err)
+	}
+
+	extraInitContainerCustomFileCopyMap := map[string]any{
+		"name":            "copy-custom-files",
+		"image":           comp.Spec.Parameters.Service.CustomizationImage.Image,
+		"imagePullPolicy": "Always",
+		"command": []string{
+			"sh",
+		},
+		"args": []string{
+			"-c",
+			copyCommand.String(),
+		},
+		"volumeMounts": []map[string]string{{
+			"name":      "customization-image-files",
+			"mountPath": "/custom",
+		}},
+	}
+	extraInitContainersMap = append(extraInitContainersMap, extraInitContainerCustomFileCopyMap)
+
+	return extraInitContainersMap, nil
+}
+
+func copyConfigMap(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime) (*corev1.ConfigMap, error) {
+
+	cmObjectName := comp.GetName() + "-config-claim-observer"
+
+	cmClaimObserver := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      *comp.Spec.Parameters.Service.CustomConfigurationRef,
+			Namespace: comp.GetClaimNamespace(),
+		},
+	}
+	err := svc.SetDesiredKubeObject(cmClaimObserver, cmObjectName, runtime.KubeOptionObserve)
+
+	if err != nil {
+		svc.Log.Info("Can't observe Keycloak config Configmap")
+		svc.AddResult(runtime.NewFatalResult(fmt.Errorf("cannot get configMap: %w", err)))
+	}
+
+	configMapClaim := &corev1.ConfigMap{}
+	err = svc.GetObservedKubeObject(configMapClaim, cmObjectName)
+	if err != nil {
+		svc.Log.Info("cannot get observed Keycloak config map")
+		svc.AddResult(runtime.NewWarningResult("cannot get observed Keycloak config map"))
+	}
+
+	configMapInstance := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        *comp.Spec.Parameters.Service.CustomConfigurationRef,
+			Namespace:   comp.GetInstanceNamespace(),
+			Labels:      configMapClaim.Labels,
+			Annotations: configMapClaim.Annotations,
+		},
+		Data: configMapClaim.Data,
+	}
+
+	return configMapInstance, svc.SetDesiredKubeObject(configMapInstance, comp.GetName()+"-config-map")
+}
+
+func copySecret(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime) error {
```

Comment:
`copySecret` and `copyConfigMap` have been removed in favor of `CopyKubeResource`
I think you have to rebase